File: src/Deviser.Core/Deviser.Core.Common/DomainTypes/LayoutType.cs
﻿using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace Deviser.Core.Common.DomainTypes
{
    public class LayoutType
    {
        public Guid Id { get; set; }
        public string Name { get; set; } //old property - Type
        public string Label { get; set; }
        public string IconImage { get; set; }
        public string IconClass { get; set; }
        public string LayoutTypeIds { get; set; }
        public ICollection<Property> Properties { get; set; }
        public ICollection<LayoutType> AllowedLayoutTypes { get; set; }
        public bool IsActive { get; set; }
        public string IsActiveText => IsActive ? "Active" : "In Active";
        public string IsActiveBadgeClass => IsActive ? "badge-primary" : "badge-secondary";
        public DateTime? CreatedDate { get; set; }
        public DateTime? LastModifiedDate { get; set; }
        
    }
}

File: src/Deviser.Web/ClientApp/src/app/common/domain-types/layout-type.ts
import { Property } from './property';

export interface LayoutType {
    id: string;
    name: string; //old property - Type
    label: string;
    iconImage: string;
    iconClass: string;
    layoutTypeIds: string;
    properties: Property[];
    allowedLayoutTypes: LayoutType[];
    isActive: boolean;
    //IsActiveText: string;
    isActiveBadgeClass: string;
    createdDate: Date | string | null;
    lastModifiedDate: Date | string | null;
}